src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -295,18 +323,73 @@ var MarkdownHighlightRules = function() {
             token : "comment",
             regex : "<\\!--",
             next  : "html-comment"
-        }, {
+        },{
             include : "basic"
         }],
 
+        "cm-insertion" : [{
+           token: "criticmarkup_insertion",
+           regex: "\\+\\+\\}",
+           next: "start"
```

Comment:
Re: previous comment; this may be better as `next: "pop"`. Similar for other usages of `next: "start"`.